homeassistant/components/elmax/config_flow.py

```diff
@@ -0,0 +1,249 @@
+            )
+
+        except ElmaxBadLoginError:
+            _LOGGER.error("Wrong credentials or failed login")
+            errors["base"] = "bad_auth"
+        except NoOnlinePanelsError:
+            _LOGGER.warning("No online device panel was found")
+            errors["base"] = "no_panel_online"
+        except ElmaxNetworkError:
+            _LOGGER.exception("A network error occurred")
+            errors["base"] = "network_error"
+
+        # If an error occurred, show back the login form.
+        return self.async_show_form(
+            step_id="user", data_schema=LOGIN_FORM_SCHEMA, errors=errors
+        )
+
+    async def async_step_panels(self, user_input: dict[str, Any]) -> FlowResult:
+        """Handle Panel selection step."""
+        errors = {}
+        panel_name = user_input[CONF_ELMAX_PANEL_NAME]
+        panel_pin = user_input[CONF_ELMAX_PANEL_PIN]
+
+        # Lookup the panel id from the panel name.
+        panel_id = self._panel_names[panel_name]
+
+        # Make sure this is the only elmax integration for this specific panel id.
+        await self.async_set_unique_id(panel_id)
+        self._abort_if_unique_id_configured()
+
+        # Try to list all the devices using the given PIN.
+        try:
+            await self._client.get_panel_status(
+                control_panel_id=panel_id, pin=panel_pin
+            )
+            return self.async_create_entry(
+                title=f"Elmax {panel_name}",
+                data={
+                    CONF_ELMAX_PANEL_ID: panel_id,
+                    CONF_ELMAX_PANEL_PIN: panel_pin,
+                    CONF_ELMAX_USERNAME: self._username,
+                    CONF_ELMAX_PASSWORD: self._password,
+                },
+            )
+        except ElmaxBadPinError:
+            errors["base"] = "invalid_pin"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Error occurred")
+            errors["base"] = "unknown_error"
+
+        return self.async_show_form(
+            step_id="panels", data_schema=self._panels_schema, errors=errors
+        )
+
+    async def async_step_reauth(self, user_input=None):
+        """Perform reauth upon an API authentication error."""
+        self._reauth_username = user_input.get(CONF_ELMAX_USERNAME)
+        self._reauth_panelid = user_input.get(CONF_ELMAX_PANEL_ID)
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(self, user_input=None):
+        """Handle reauthorization flow."""
+        errors = {}
+        if user_input is not None:
+            panel_pin = user_input.get(CONF_ELMAX_PANEL_PIN)
+            password = user_input.get(CONF_ELMAX_PASSWORD)
+            entry = await self.async_set_unique_id(self._reauth_panelid)
+
+            # Handle authentication, make sure the panel we are re-authenticating against is listed among results
+            # and verify its pin is correct.
+            try:
```

Comment:
There is much stuff in this try/except scope that doesn't throw, please move those part out of this scope.